crates/red_knot_python_semantic/src/python_version.rs

```diff
@@ -0,0 +1,136 @@
+use std::fmt;
+
+/// Enumeration of all supported Python versions
+///
+/// TODO: unify with the `PythonVersion` enum in the linter/formatter crates?
+#[derive(Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Default)]
+pub enum TargetVersion {
+    Py37,
+    #[default]
+    Py38,
+    Py39,
+    Py310,
+    Py311,
+    Py312,
+    Py313,
+}
+
+impl TargetVersion {
+    pub fn major_version(self) -> u8 {
+        PythonVersion::from(self).major
+    }
+
+    pub fn minor_version(self) -> u8 {
+        PythonVersion::from(self).minor
+    }
+
+    const fn as_display_str(self) -> &'static str {
+        match self {
+            Self::Py37 => "py37",
+            Self::Py38 => "py38",
+            Self::Py39 => "py39",
+            Self::Py310 => "py310",
+            Self::Py311 => "py311",
+            Self::Py312 => "py312",
+            Self::Py313 => "py313",
+        }
+    }
+}
+
+impl fmt::Display for TargetVersion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.as_display_str())
+    }
+}
+
+impl fmt::Debug for TargetVersion {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(self, f)
+    }
+}
+
+/// Generic representation for a Python version.
+///
+/// Unlike [`TargetVersion`], this does not necessarily represent
+/// a Python version that we actually support.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct PythonVersion {
```

Comment:
We don't need to make this change in this PR but I think it's now a bit confusing when to use `TargetVersion` vs `PythonVersion`? Shouldn't all (or at least most) our code be open for new Python versions? Or to phrase it differently, what's the benefit of limiting `TargetVersion` to a fixed set? 

To me, the main reason why `TargetVersion` is an enum is just that it was easier to reference specific versions. But I think that can also be accomplished by `PythonVersion`, we can define constants for the versions that we care about (and we can implement `Default`).

Reply:
Hmm, maybe. Having it as an enum is also useful for command-line parsing. But you might be right that we could just use the `PythonVersion` type everywhere. Not sure; worth experimenting with as a followup, anyway!